### PR TITLE
Fixes #13181 - continue uploading reports when one fails

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

File: extra/smart-proxy-openscap-send.cron [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # Send all collected OpenSCAP reports once every 30 minutes
-*/30 * * * * foreman-proxy smart-proxy-openscap-send >>/var/log/foreman-proxy/cron.log 2>&1
+*/30 * * * * foreman-proxy smart-proxy-openscap-send >>/var/log/foreman-proxy/cron.log > /dev/null 2>&1
```
Comment (ares) [inline extra/smart-proxy-openscap-send.cron]: this seems unrelated, also I'd say I want  to get the output if something goes wrong

File: lib/smart_proxy_openscap/spool_forwarder.rb [hunk 2]
```diff
@@ -3,11 +3,19 @@ class SpoolForwarder
     include ::Proxy::Log
 
     def post_arf_from_spool(arf_dir)
+      f = nil
```
Comment (ares) [inline lib/smart_proxy_openscap/spool_forwarder.rb]: could you please rename `f` to `failed` to be more explicit?

File: bin/smart-proxy-openscap-send [hunk 3]
```diff
@@ -37,8 +37,8 @@ end
 
 begin
   Proxy::OpenSCAP::send_spool_to_foreman
-rescue StandardError => e
-  logger.debug e.backtrace.join("\n\t")
-  logger.error "Failed to send SCAP results to the Foreman server: #{e}"
+rescue RuntimeError => e
```
Comment (ares) [inline bin/smart-proxy-openscap-send]: why is this changed to RuntimeError? I see that StandardError is being catched in spool_forwarder.rb now but if it's risen somewhere else we should still catch them (e.g. IOError).

File: extra/smart-proxy-openscap-send.cron (hunk 1 above)
Comment (sean797) [inline extra/smart-proxy-openscap-send.cron]: Yes, that would make more sense :)

**Comment by sean797** [comment]
@ares Made those changes!

File: lib/smart_proxy_openscap/spool_forwarder.rb [hunk 4]
```diff
@@ -3,11 +3,19 @@ class SpoolForwarder
...
+          next if cname == '.' || cname == '..'
+          cname_dir = File.join(arf_dir, cname)
+          forward_cname_dir(cname, cname_dir) if File.directory?(cname_dir)
+        rescue StandardError => e
+          logger.debug e.backtrace.join("\n\t") 
+          logger.error "Failed to send SCAP results for #{cname} to the Foreman server: #{e}" 
+          failed = true
+        end
       end
+      raise "Failed to send SCAP results for one or more hosts." if f
```
Comment (ares) [inline lib/smart_proxy_openscap/spool_forwarder.rb]: `if failed` I guess

File: bin/smart-proxy-openscap-send [hunk 5]
```diff
@@ -37,8 +37,8 @@ end
 
 begin
   Proxy::OpenSCAP::send_spool_to_foreman
-rescue StandardError => e
-  logger.debug e.backtrace.join("\n\t")
-  logger.error "Failed to send SCAP results to the Foreman server: #{e}"
+rescue StandardError, RuntimeError => e
```
Comment (ares) [inline bin/smart-proxy-openscap-send]: `StandardError` will catch `RuntimeError` as well (it inherits from it) so it doesn't have to be mentioned here

File: lib/smart_proxy_openscap/spool_forwarder.rb (hunk 4 above)
Comment (sean797) [inline lib/smart_proxy_openscap/spool_forwarder.rb]: sorry that was really stupid of me...

File: bin/smart-proxy-openscap-send (hunk 5 above)
Comment (sean797) [inline bin/smart-proxy-openscap-send]: Okay, I'm fairly new to Ruby - Thanks!

File: lib/smart_proxy_openscap/spool_forwarder.rb (hunk 4 above)
Comment (ares) [inline lib/smart_proxy_openscap/spool_forwarder.rb]: no worries ;-)

File: extra/smart-proxy-openscap-send.cron [hunk 6]
```diff
@@ -1,2 +1,2 @@
 # Send all collected OpenSCAP reports once every 30 minutes
-*/30 * * * * foreman-proxy smart-proxy-openscap-send >>/var/log/foreman-proxy/cron.log 2>&1
+*/30 * * * * foreman-proxy smart-proxy-openscap-send >>/var/log/foreman-proxy/cron.log > /dev/null
```
Comment (ares) [inline extra/smart-proxy-openscap-send.cron]: last comment - I think you can remove `> /dev/null`, we already redirect STDOUT to cron.log

Comment (shlomizadok) [inline extra/smart-proxy-openscap-send.cron]: one last nitpick: spacing between the `>>` to `/var/log...`

**Comment by shlomizadok** [comment]
@sean797 - Thanks for this important addition. Can you please just fix two small nitpicks so we can merge? Thanks :)

**Comment by sean797** [comment]
@shlomizadok - Done :)

**Comment by ares** [comment]
ACK :+1:

**Comment by shlomizadok** [comment]
@sean797 - Thank you for this fix!
@ares, thanks for testing.
